CONTRIBUTING.md

```diff
@@ -32,7 +32,60 @@ When requesting or submitting new features, first consider whether it might be u
 
 - Submit Pull Requests to the **development branch only**.
 - Before Submitting your Pull Request, merge `development` with your new branch and fix any conflicts. (Make sure you don't break anything in development!)
-- Please use the [Google Style Guide for Shell](https://google.github.io/styleguide/shell.xml) for your code submission styles. 
+- Please use the [Google Style Guide for Shell](https://google.github.io/styleguide/shell.xml) for your code submission styles.
 - Commit Unix line endings.
 - Please use the Pi-hole brand: **Pi-hole** (Take a special look at the capitalized 'P' and a low 'h' with a hyphen)
 - (Optional fun) keep to the theme of Star Trek/black holes/gravity.
+
+## Forking and Cloning from GitHub to GitHub
+
+1. Fork <https://github.com/pi-hole/pi-hole/> to a repo under a namespace you control, or have permission to use, example: [https://github.com/<your_namespace>/pi-hole/]().  You can do this from the github.com website.
+2. Clone [https://github.com/<your_namespace>/pi-hole/]() with the tool of you choice.
+3. To keep your fork in sync with our repo, add an upstream remote for pi-hole/pi-hole to your repo.
+    ```console
+    git remote add upstream https://github.com/pi-hole/pi-hole.git
+    ```
+4. Checkout the `development` branch from your clone [https://github.com/<your_namespace>/pi-hole/]().
+5. Create a topic/branch, based on the `development` branch code. *Bonus fun to keep to the theme of Star Trek/black holes/gravity.*
+6. Make your changes and commit to your topic branch in your repo.
+7. rebase your commits and squash any insignificant commits.  See notes below for an example.
+8. Merge `development` your branch and fix any conflicts.
+9. Open a Pull Request to merge your topic branch into our repo's `development` branch.
+- Keep in mind the technical requirements from above.
+
+## Forking and Cloning from GitHub to other code hosting sites
+
+- Forking is a GitHub concept and cannot be done from GitHub to other git based code hosting sites.  However, from those sites may be able to mirror a GitHub repo.
+1. To contribute from another code hosting site, you must first complete the steps above to fork our repo to a GitHub namespace you have permission to use, example: [https://github.com/<your_namespace>/pi-hole/]()
+2. Create a repo in your code hosting site, for example: [https://gitlab.com/<your_namespace>/pi-hole/]()
+3. Follow the instructions from your code hosting site to create a mirror between [https://github.com/<your_namespace>/pi-hole/]() and [https://gitlab.com/<your_namespace>/pi-hole/]().
+4. When you are ready to create a Pull Request (PR), follow the steps `(starting at step #6)` from [Forking and Cloning from GitHub to GitHub](#forking-and-cloning-from-github-to-github) and create the PR from [https://github.com/<your_namespace>/pi-hole/]().
```

Comment:
Would be easier to just `git clone --mirror` and then attach the local mirror to what ever service you wish to host it on?

https://stackoverflow.com/a/3960063

Reply:
That's step #3.  You have to have write permission to both repos to mirror.   For a push mirror, every commit is mirrored, there are no PR between them.  If you do a pull mirror, none of the commits will be mirrored back.